examples/image_embedder/ios/ImageEmbedder/Services/ImageEmbedderService.swift

```diff
@@ -0,0 +1,243 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import MediaPipeTasksVision
+import AVFoundation
+
+/**
+ This protocol must be adopted by any class that wants to get the embedding results of the image embedder in live stream mode.
+ */
+protocol ImageEmbedderServiceLiveStreamDelegate: AnyObject {
+  func imageEmbedderService(_ imageEmbedderService: ImageEmbedderService,
+                             didFinishEmbedding result: ResultBundle?,
+                             error: Error?)
+}
+
+/**
+ This protocol must be adopted by any class that wants to take appropriate actions during  different stages of image embedding on videos.
+ */
+protocol ImageEmbedderServiceVideoDelegate: AnyObject {
+ func imageEmbedderService(_ imageEmbedderService: ImageEmbedderService,
+                                  didFinishEmbeddingOnVideoFrame index: Int)
+ func imageEmbedderService(_ imageEmbedderService: ImageEmbedderService,
+                             willBeginEmbedding totalframeCount: Int)
+}
+
+
+// Initializes and calls the MediaPipe APIs for embedding.
+class ImageEmbedderService: NSObject {
+
+  weak var liveStreamDelegate: ImageEmbedderServiceLiveStreamDelegate?
+  weak var videoDelegate: ImageEmbedderServiceVideoDelegate?
+
+  var imageEmbedder: ImageEmbedder?
+  private(set) var runningMode: RunningMode
+  private var modelPath: String
+  private var delegate: ImageEmbedderDelegate
+
+  // MARK: - Custom Initializer
+  private init?(model: Model,
+                runningMode:RunningMode,
+                delegate: ImageEmbedderDelegate) {
+    guard let modelPath = model.modelPath else { return nil }
+    self.modelPath = modelPath
+    self.runningMode = runningMode
+    self.delegate = delegate
+    super.init()
+
+    createImageEmbedder()
+  }
+
+  private func createImageEmbedder() {
+    let imageEmbedderOptions = ImageEmbedderOptions()
+    imageEmbedderOptions.runningMode = runningMode
+    imageEmbedderOptions.baseOptions.modelAssetPath = modelPath
+    imageEmbedderOptions.baseOptions.delegate = delegate.delegate
+    if runningMode == .liveStream {
+      imageEmbedderOptions.imageEmbedderLiveStreamDelegate = self
+    }
+    do {
+      imageEmbedder = try ImageEmbedder(options: imageEmbedderOptions)
+    }
+    catch {
+      print(error)
+    }
+  }
+
+  // MARK: - Static Initializers
+  static func videoImageEmbedderService(
+    model: Model,
+    videoDelegate: ImageEmbedderServiceVideoDelegate?,
+    delegate: ImageEmbedderDelegate) -> ImageEmbedderService? {
+    let imageEmbedderService = ImageEmbedderService(
+      model: model,
+      runningMode: .video,
+      delegate: delegate)
+    imageEmbedderService?.videoDelegate = videoDelegate
+
+    return imageEmbedderService
+  }
+
+  static func liveStreamEmbedderService(
```

Comment:
Rather than having three functions that essentially do the same thing, can we pass in a value to say 'use live stream, image, or video' and then do the one-off operations based on that inside of the same function?

Reply:
Because it is an example, I want to make it clear and easy to use for each case.
it is similar to the other examples in the mediapipe task

Reply:
We might want to revisit what we're doing in the other iOS examples then. Across Android we have one 'setup' function with flags.

https://github.com/googlesamples/mediapipe/blob/main/examples/face_landmarker/android/app/src/main/java/com/google/mediapipe/examples/facelandmarker/FaceLandmarkerHelper.kt#L70